File: Python developer/step07/7.02/7.2.05.py
'''

Вашей программе на вход подается массив содержащий объекты со структурой ниже.

{
    "id": 1,
    "parents": [1,2,3,4],
    "chief": {
        "name": "Paul",
        "age": 50
    },
    "age": 22
}

Определите максимальное значение во всех полях age этого объекта.

Ввод:

d – json-объект

Вывод:

res – максимальный возраст

Sample Input 1:

[{"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 50},"age": 22}, {"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 62},"age": 80}]

Sample Output 1:

80

Sample Input 2:

[{"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 100},"age": 22}, {"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 62},"age": 80}]

Sample Output 2:

100

Sample Input 3:

[{"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 50},"age": 100}, {"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 62},"age": 80}]

Sample Output 3:

100

Sample Input 4:

[{"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 50},"age": 22}, {"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 100},"age": 80}]

Sample Output 4:

100

Sample Input 5:

[{"id": 1,"parents": [1,2,3,4],"chief": {"name": "Paul","age": 50},"age": 22}]

Sample Output 5:

50

'''

import json
d = json.loads(input())

#ваш код (используйте переменную d)
l = []
for i in range(len(d)):
    l.append(d[i]['age'])
    l.append(d[i]['chief']['age'])

print(max(l))

File: Fundamentals of C programming/step07/7.2/7.2.05.c
/*

Максимум и минимум
Необходимо найти максимальный и минимальный элемент последовательности, неизвестной длины.

Входные данные:
Последовательность целых чисел, оканчивающаяся нулём. Само число нуль не является членом последовательности, а является лишь сигналом того, что достигнут конец последовательности. В последовательности есть как минимум одно ненулевое число.

Выходные данные:
Вывести два целых числа MM и mm, где MM - максимальный элемент последовательности, mm - минимальный элемент последовательности.

Sample Input 1:

123 443 652 828 2 4212 0

Sample Output 1:

4212 2

Sample Input 2:

3 0

Sample Output 2:

3 3

*/

#include <stdio.h>

int main(void) {

  int min=9999,max=-9999,number = 1;
  scanf("%d", &number);

  while (number!=0) {
    
    if (number >= max)
        max = number;
    if (min >= number)
        min = number;
	scanf("%d", &number);
  }

  printf("%d %d\n", max, min);

  return 0;
}
